lib/improver/tests/spotdata/spotdata/test_extrema.py

```diff
@@ -0,0 +1,328 @@
+        self.assertEqual(result[1].coord('time').bounds[0, 0], [lower_bound])
+        self.assertEqual(result[1].coord('time').bounds[0, 1], [upper_bound])
+
+    def test_extrema_values_day1(self):
+        """
+        Test the actual values returned by the collapse method to ensure it is
+        successfully extracting the maximum/minimum temperatures in the
+        defined period.
+
+        UTC times : 00  01  02  ... 10 11 12 13 14
+        UTC offset:-12 -11 -10  ... -2 -1  0  1  2
+        site index:  0   1   2  ... 10 11 12 13 14
+        Local time: 12  13  14  ... 22 23 00 01 02
+
+        site_index 12 at time_index 0 should be adjusted to fall at 00 27th
+        April 2017 in local time, so is expected to fall outside day 1. Thus
+        setting a high value for this site should leave the maximum unset for
+        site 12.
+
+        site_index 2 at time_index 9 (09Z 27 April 2017) will fall at 23 26th
+        April 2017 local time, so setting this to 40 should modify the maximum
+        for site 2 on day 1.
+
+        """
+        self.cube.data[9, 2] = 40
+        self.cube.data[0, 12] = 40
+
+        # Expected data array.
+        expected = np.arange(0, 27).astype(float)
+        expected[2] = 40.
+
+        result = Plugin(24, start_hour=0).process(self.cube)
+        result = result.extract(Constraint(name='air_temperature_max'))
+        self.assertTrue(result[0].data[12].mask)
+        self.assertArrayEqual(result[0].data, expected)
+
+
+class Test_get_valid_dates(Test_extrema):
+    """Test extraction of day min and max and hour setting."""
+
+    def test_get_valid_dates_6(self):
+        """
+        Given an iris time coord, check that this function returns the day
+        min and maxima with the provided hour appended.
+
+        """
+        expected_start = dt(2017, 03, 27, 6)
+        expected_end = dt(2017, 03, 29, 6)
+        result_start, result_end = get_valid_dates(self.time_coord,
+                                                   start_hour=6)
+        self.assertEqual(expected_start, result_start)
+        self.assertEqual(expected_end, result_end)
+
+    def test_get_valid_dates_non_int_hour(self):
+        """
+        Check an error is raised if a non-integer hour is provided.
+
+        """
+        msg = "integer argument expected, got float"
+        with self.assertRaisesRegexp(TypeError, msg):
+            get_valid_dates(self.time_coord, start_hour=6.2)
+
+
+class Test_make_local_time_cube(Test_extrema):
+    """Test time localisation function."""
+
+    def test_time_coord(self):
+        """
+        Test that a new time_coord has been constructed that spans all local
+        times. UTC-12 to UTC+14 is the maximum range of timezones.
+
+        """
+        expected_first_time = self.time_coord.points[0] - 12
+        expected_last_time = self.time_coord.points[-1] + 14
+        result = make_local_time_cube(self.cube)
```

Comment:
Index is just a monotonically increasing index that is necessary to have any kind of dimension coordinate. The cubes going in here are already spotdata cubes and so have this index coordinate. I have removed the setting of a new coordinate in make_local_time_cube and recycled the coordinate from the input cube as they should always be the same size.